src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/KubernetesDeclarativeAgent.java

```diff
@@ -26,11 +37,12 @@
     private int activeDeadlineSeconds;
 
     private ContainerTemplate containerTemplate;
+    private PodTemplate podTemplate;
+    private String podTemplateName;
 
     @DataBoundConstructor
-    public KubernetesDeclarativeAgent(String label, ContainerTemplate containerTemplate) {
+    public KubernetesDeclarativeAgent(String label) {
```

Comment:
I figured if a `KubernetesDeclarativeAgent` could be created with either a ContainerTemplate or a PodTemplate then we'd need to make them optional via setters rather than making the ContainerTemplate mandatory in the constructor? 

If we are creating just a PodTemplate in declarative will it pass in null for the ContainerTemplate anyways? (Am still finding my feet with declarative ;)

Reply:
you could do 

```
@deprecated
public KubernetesDeclarativeAgent(String label, ContainerTemplate containerTemplate) {...

@DataBoundConstructor
public KubernetesDeclarativeAgent(String label) {...

@databoundsetter
public setContainerTemplate(...
```

Reply:
ah great idea thanks!